packages/react-select/src/components/LiveRegion.js

```diff
@@ -0,0 +1,197 @@
+  const {
+    inputValue,
+    ariaSelection,
+    focusedOption,
+    focusedValue,
+    focusableOptions,
+    isFocused,
+    menuIsOpen,
+    selectValue,
+    options,
+    selectProps,
+  } = props;
+
+  const {
+    ariaLiveMessages,
+    getOptionLabel,
+    isDisabled,
+    isMulti,
+    isOptionDisabled,
+    isSearchable,
+    screenReaderStatus,
+    tabSelectsValue,
+  } = selectProps;
+  const ariaLabel = selectProps['aria-label'];
+  const ariaLive = selectProps['aria-live'];
+
+  const [ariaSelected, setAriaSelected] = useState('');
+  const [ariaFocused, setAriaFocused] = useState('');
+  const [ariaResults, setAriaResults] = useState('');
+  const [ariaGuidance, setAriaGuidance] = useState('');
+  const [messages, setMessages] = useState(defaultAriaLiveMessages);
+
+  // Update aria live message configuration when prop changes
+  useEffect(
+    function updateAriaLiveMessages() {
+      setMessages({
+        ...defaultAriaLiveMessages,
+        ...(ariaLiveMessages || {}),
+      });
+    },
+    [ariaLiveMessages]
+  );
+
+  // Update aria live selected option when prop changes
+  useEffect(
+    function onChange() {
+      let message = '';
+      if (ariaSelection && messages.onChange) {
+        const { value, ...context } = ariaSelection;
+        message = messages.onChange(value, { ...context });
+      }
+      setAriaSelected(message);
+    },
+    [ariaSelection, messages]
+  );
+
+  // Formatting function for newly focused option/value
+  useEffect(
+    function onFocus() {
+      let focusMsg = '';
+      const focused = focusedOption || focusedValue;
+      const isSelected = !!(
+        focusedOption &&
+        selectValue &&
+        selectValue.includes(focusedOption)
+      );
+
+      if (focused && messages.onFocus) {
+        const context = {
+          label: getOptionLabel(focused),
+          isDisabled: isOptionDisabled(focused),
+          isSelected,
+          options,
+          type: focused === focusedOption ? 'option' : 'value',
+          value: selectValue,
+        };
+
+        focusMsg = messages.onFocus(focused, { ...context });
+      }
+      setAriaFocused(focusMsg);
+    },
+    [
+      focusedOption,
+      focusedValue,
+      getOptionLabel,
+      isOptionDisabled,
+      menuIsOpen,
+      messages,
+      options,
+      selectValue,
+    ]
+  );
+
+  useEffect(
+    function onFilter() {
+      let resultsMsg = '';
+      if (menuIsOpen && options.length && messages.onFilter) {
+        const resultsMessage = screenReaderStatus({
+          count: focusableOptions.length,
+        });
+        resultsMsg = messages.onFilter({ inputValue, resultsMessage });
+      }
+      setAriaResults(resultsMsg);
+    },
+    [
+      focusableOptions,
+      inputValue,
+      menuIsOpen,
+      messages,
+      options,
+      screenReaderStatus,
+    ]
+  );
+
+  useEffect(
+    function guidance() {
+      let guidanceMsg = '';
+      if (messages.guidance) {
```

Comment:
do we need to check `messages.guidance` if `defaultAriaLiveMessages` is spread into `messages`?

Reply:
@bozdoz Good question with a few thoughts here:

The thought here was if a user wanted to remove any of the messages, they could pass in 
```jsx
ariaLiveMessages={ guidance: null }
```

Guidance in particular can get a bit wordy and the default behavior is such that screen readers will read both this and the listbox or textbox instructions.

I also envisioned some advanced level of customization where a function can return false or null if some criteria is met. In this case, if the user is navigating a listbox, they may not want to hear the directions after some amount of time, interaction, etc... so the thinking was to make this as dynamic as possible.

Reply:
`guidance: null` looks good to me; the types did not make that API apparent to me.  Perhaps you could change the functions to be Maybe types (as opposed to optional):

```diff
- guidance?: (type: GuidanceType, context?: GuidanceContext) => string,
+ guidance?: ?(type: GuidanceType, context?: GuidanceContext) => string,
```

That way `messages.guidance` can be null